lldb/source/Commands/CommandObjectScripting.cpp

```diff
@@ -127,6 +130,171 @@ class CommandObjectScriptingRun : public CommandObjectRaw {
+                                         option_arg.str().c_str());
+        break;
+      default:
+        llvm_unreachable("Unimplemented option");
+      }
+
+      return error;
+    }
+
+    void OptionParsingStarting(ExecutionContext *execution_context) override {
+      language = lldb::eScriptLanguageNone;
+    }
+
+    llvm::ArrayRef<OptionDefinition> GetDefinitions() override {
+      return llvm::ArrayRef(g_scripting_template_list_options);
+    }
+
+    lldb::ScriptLanguage language = lldb::eScriptLanguageNone;
+  };
+
+protected:
+  void DoExecute(Args &command, CommandReturnObject &result) override {
+    lldb::ScriptLanguage language =
+        (m_options.language == lldb::eScriptLanguageNone)
+            ? m_interpreter.GetDebugger().GetScriptLanguage()
+            : m_options.language;
+
+    if (language == lldb::eScriptLanguageNone) {
+      result.AppendError(
+          "the script-lang setting is set to none - scripting not available");
+      return;
+    }
+
+    ScriptInterpreter *script_interpreter =
+        GetDebugger().GetScriptInterpreter(true, language);
+
+    if (script_interpreter == nullptr) {
+      result.AppendError("no script interpreter");
+      return;
+    }
+
+    Stream &s = result.GetOutputStream();
+    s.Printf("Available scripted affordances:\n");
+
+    auto print_field = [&s](llvm::StringRef key, llvm::StringRef value,
+                            bool check_validy = false) {
+      if (!check_validy || !value.empty()) {
+        s.IndentMore();
+        s.Indent();
+        s << key << ": " << value << '\n';
+        s.IndentLess();
+      }
+    };
+    auto print_usages = [&s](llvm::StringRef usage_kind,
+                             std::vector<llvm::StringRef> &usages) {
+      s.IndentMore();
+      s.Indent();
+      s << usage_kind << " Usages:";
+      if (usages.empty())
+        s << " No usages.\n";
+      else if (usages.size() == 1)
+        s << " " << usages.front() << '\n';
+      else {
+        s << '\n';
+        for (llvm::StringRef usage : usages) {
+          s.IndentMore();
+          s.Indent();
+          s << usage << '\n';
+          s.IndentLess();
+        }
+      }
+      s.IndentLess();
+    };
+
+    size_t i = 0;
+    for (llvm::StringRef plugin_name =
+             PluginManager::GetScriptedInterfaceNameAtIndex(i);
+         !plugin_name.empty();) {
+
+      llvm::StringRef desc =
+          PluginManager::GetScriptedInterfaceDescriptionAtIndex(i);
+      lldb::ScriptLanguage lang =
+          PluginManager::GetScriptedInterfaceLanguageAtIndex(i);
+      std::vector<llvm::StringRef> ci_usages =
+          PluginManager::GetScriptedInterfaceCommandInterpreterUsagesAtIndex(i);
+      std::vector<llvm::StringRef> api_usages =
+          PluginManager::GetScriptedInterfaceAPIUsagesAtIndex(i);
+
+      print_field("Name", plugin_name);
+      switch (lang) {
+      case eScriptLanguagePython:
+        print_field("Language", "Python");
```

Comment:
Use ScriptInterpreter::LanguageToString rather than hard coding the values here.